tests/e2e/k8s/services/httpecho/client.go

```diff
@@ -77,10 +79,22 @@ func GetEchoValue(cluster *util.KindCluster, server *util.Service) (string, erro
 
 func RunClientInPod(cluster *util.KindCluster, server *util.Service) (string, error) {
 	body, err := cluster.RunPod(&util.Pod{
-		Name:      "echo-client",
+		Name:      EchoClientPodName,
 		Namespace: server.Namespace,
 		Image:     "curlimages/curl",
 		Args:      []string{"curl", "-s", "-m", "1", "http://" + server.Name},
 	})
 	return strings.TrimSpace(body), err
 }
+
+// Sleep allows more time for CL to be notified of the new Pod, so CL can retrieve the Pod's info.
+func RunClientInPodWithSleep(cluster *util.KindCluster, server *util.Service) (string, error) {
```

Comment:
I don't think you need this function. If the non-sleep function fails, you will be covered by the `allowRetry=true` for the `AccessService` function.

Reply:
How does the retry work when we run the client in a Pod?
Aren't we just creating Pods again and again?

Reply:
Right, my bad.
So maybe add a flag that removes the `RestartPolicy: v1.RestartPolicyNever`.

Reply:
> Right, my bad. So maybe add a flag that removes the `RestartPolicy: v1.RestartPolicyNever`.

This won't work.

How about using
```--retry-delay 1 --retry-all-errors```
in the curl pod command line?

Reply:
yes, this works.